### ISSUE TITLE
[Banner] Heading id is read by screen reader instead of the actual heading content 

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react#1931]
## Issue summary

When using a screen reader, the heading `id` that gets set on the `Banner` `aria-labelledby` attribute is read instead of the heading's content. I believe this is because the `Banner` heading `id` is set on the wrapping `div` instead of the semantic content element that `Heading` renders.

## Expected behavior

The banner heading content is read followed by the banner content.

## Actual behavior

The banner heading's `id` is read followed by the banner content.

## Steps to reproduce the problem

1. `git checkout master && yarn dev` to open the Playground/Storybook locally
2. Use a screen reader to read any `Banner` example

**Comment by chloerice** [comment]
cc @s-johnston

**Comment by chloerice** [comment]
I've been unable to reproduce this with VoiceOver. This issue was raised by @s-johnston, so I've reached out to her for more specifics on the browser etc that this issue was experienced in.
